### ISSUE TITLE
etl::make_string fails for empty strings

### ISSUE THREAD

**Issue by twam** [ETLCPP/etl#299]
With 18.19.2 creating an empty string with etl::make_string like

```
#include <etl/string.h>

int main() {
    const auto emptyString = etl::make_string("");
}
```

fails (see https://godbolt.org/z/WvYhdT).

This was working fine with 17.7.1.

**Comment by jwellbelove** [comment]
Yes, the issue has occurred since strings with external buffers were introduced in 18.1.0.
make_string("") tries to create a string of length 0, which now indicates that the string has an external buffer.
I'll push a fix.

**Comment by jwellbelove** [comment]
The fix does have the side effect of `max_size()` returning `1` instead of `0`.

**Comment by jwellbelove** [comment]
Actually, I have a fix now that has no side effects.

**Comment by jwellbelove** [comment]
Fixed in 18.20.0
